packages/atomic/src/utils/store.ts

```diff
@@ -0,0 +1,27 @@
+import {NumericFacetValue, DateFacetValue} from '@coveo/headless';
+
+interface FacetLabel {
+  label: string;
+}
+
+interface FacetValueFormat<ValueType> {
+  format(facetValue: ValueType): string;
+}
+
+type FacetStore<F extends FacetLabel> = Record<string, F>;
+
+export type AtomicStore = {
+  facets: FacetStore<FacetLabel>;
+  numericFacets: FacetStore<FacetLabel & FacetValueFormat<NumericFacetValue>>;
+  dateFacets: FacetStore<FacetLabel & FacetValueFormat<DateFacetValue>>;
+  categoryFacets: FacetStore<FacetLabel>;
+  refineEnabled: boolean;
```

Comment:
Why should this state be centralized in the store vs. being localized in the refine toggle component?

Reply:
Because it affects another component can is outside of itself and not a parent of it (then custom events could be used).
I'll see... maybe if I inject atomic-refine-modal I can keep a reference to it in refine-toggle and toggle it this way.